tpmtool.cpp

```diff
@@ -626,139 +628,246 @@ CreateSpace (
     return 0;
 }
 
-int32_t
-QuerySpace (
-    int32_t ArgumentCount,
+TPM_RC
+QuerySpaceInternal (
     uintptr_t TpmHandle,
-    TPM_NV_INDEX Index
-    )
+    TPM_NV_INDEX Index,
+    uint16_t* DataSize,
+    char* Attributes,
+    char* OwnerRights,
+    char* AuthRights,
+    char* AttributeBuffer,
+    rsize_t AttributeBufferSize
```

Comment:
I notice for the other buffers, you hard-coded their size, but for AttrbuteBuffer, you pass in the size. Should pick one option :)

Reply:
Because for others I don't use their size in the internal function, so I don't need to know exactly what it is, just that it's large enough. For AttributeBufferSize I do need to use it's actual size and don't want to hard-code it inside the internal function too, in case it ever changes.

Reply:
You did use their size. You had sizeofs everywhere (not anymore since you removed the _s).